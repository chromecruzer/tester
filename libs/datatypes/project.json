{
  "name": "datatypes",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/datatypes/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/datatypes/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/datatypes"],
      "options": {
        "jestConfig": "libs/datatypes/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
