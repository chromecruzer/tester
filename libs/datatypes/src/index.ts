export * from './datatypes';
export * from './IolRecord';
export * from './CustomerRecord';
export * from './CustomerContactRecord';
export * from './ProductRecord';
export * from './EmployeeRecord';        // added by shyam for testing 
export * from './EmployeeRecordDataMappings'   //
export * from './UserRecord1'  //
export * from './UserRecordDM' // 
export * from './ConsignmentRecord';
export * from './CustomerDataMapping';
export * from './CustomerContactRecord';
export * from './CustomerContactDataMapping';
export * from './IolDataMapping';
export * from './notifications';
export * from './ProductDataMapping';
export * from './SalesRepDataMapping';
export * from './SalesRepRecord';
export * from './SalesMappingSquared';
export * from './SalesMappingRecord';
export * from './SalesJumpRecord';
export * from './UploadHeaders';
export * from './AuditRecord';
export * from './SearchSignature';
export * from './EmailSettings';
//export * from './UserRecord';
import BuildUrl from "./BuildUrl";
import {ExpirationCalculations} from "./ExpirationCalculations";
import ItemMatchStatus from './ItemMatchStatus';
import SearchSignature from "./SearchSignature";
export {BuildUrl, ExpirationCalculations, ItemMatchStatus, SearchSignature};
