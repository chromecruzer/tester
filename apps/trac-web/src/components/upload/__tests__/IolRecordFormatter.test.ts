import IolRecordFormatter from "../IolRecordFormatter";
import {DateTime} from "luxon";
import {tracDateFormat} from "@trac/datatypes";

describe('IolRecordFormatter', () => {
  describe('#formatRowValues', () => {
    it('should format the dates of an iol record', () => {
      const formatter = new IolRecordFormatter({
        adds: [],
        removes: [],
        modifies: []
      });
      const actual = formatter.formatRowValues({
        received_date: DateTime.utc(2016, 3, 26).toFormat(tracDateFormat),
        expire_date: DateTime.utc(2014, 3, 16).toFormat(tracDateFormat),
        shipped_date: DateTime.utc(2016, 3, 6).toFormat(tracDateFormat),
      });
      expect(actual).toMatchSnapshot();
    });
  });
});
