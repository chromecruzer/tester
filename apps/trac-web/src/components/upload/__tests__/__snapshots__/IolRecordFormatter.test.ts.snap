// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IolRecordFormatter #formatRowValues should format the dates of an iol record 1`] = `
Object {
  "expire_date": "03/16/2014",
  "received_date": "03/26/2016",
  "shipped_date": "03/06/2016",
}
`;
